src/main/java/com/mongodb/jdbc/MongoSQLResultSet.java

```diff
@@ -132,136 +114,65 @@ protected Object getObject(BsonValue o, int columnType) throws SQLException {
-                        return o.asDocument();
-                    case OBJECT_ID:
-                        return o.asObjectId();
-                    case DB_POINTER:
-                        return o.asDBPointer();
-                    case INT32:
-                        return o.asInt32();
-                    case INT64:
-                        return o.asInt64();
-                    case JAVASCRIPT:
-                        return o.asJavaScript();
-                    case JAVASCRIPT_WITH_SCOPE:
-                        return o.asJavaScriptWithScope();
-                    case MAX_KEY:
-                        return (BsonMaxKey) o;
-                    case MIN_KEY:
-                        return (BsonMinKey) o;
-                    case REGULAR_EXPRESSION:
-                        return (BsonRegularExpression) o;
-                    case SYMBOL:
-                        return o.asSymbol();
-                    case TIMESTAMP:
-                        return o.asTimestamp();
-                    case UNDEFINED:
-                        return (BsonUndefined) o;
-                    case NULL:
-                        return null;
-                    default:
-                        return o;
+                if (o.getBsonType() == BsonType.NULL) {
+                    return null;
                 }
-            case Types.REAL:
-                // not supported
-                break;
+                // These types are wrapped in MongoSQLBsonValue so that
+                // if they are stringified via toString() they will be
+                // represented as extended JSON.
+                return new MongoSQLBsonValue(o);
+
+            case Types.ARRAY:
+            case Types.BLOB:
+            case Types.CLOB:
+            case Types.DATALINK:
+            case Types.DATE:
+            case Types.DISTINCT:
+            case Types.JAVA_OBJECT:
+            case Types.LONGVARBINARY:
+            case Types.NCLOB:
             case Types.REF:
-                // not supported
-                break;
             case Types.REF_CURSOR:
-                // not supported
-                break;
             case Types.ROWID:
-                // not supported
-                break;
-            case Types.SMALLINT:
-                return getInt(o);
             case Types.SQLXML:
-                // not supported
-                break;
             case Types.STRUCT:
-                // not supported
-                break;
             case Types.TIME:
-                // not supported
-                break;
             case Types.TIME_WITH_TIMEZONE:
-                // not supported
-                break;
-            case Types.TIMESTAMP:
-                return getTimestamp(o);
             case Types.TIMESTAMP_WITH_TIMEZONE:
-                // not supported
-                break;
-            case Types.TINYINT:
-                return getInt(o);
             case Types.VARBINARY:
```

Comment:
Could we move VARBINARY and LONGVARBINARY with BINARY, the same way we group BIGINT, INT, SMALLINT, TINYINT and CHAR, VARCHAR, LONGVARCHAR?

Reply:
I can make this change, but does this really matter? The only JDBC types that this method will ever encounter are the ones associated with BSON types in `BsonTypeInfo`. `BsonTypeInfo` does not have any reference to `Types.VARBINARY` or `Types.LONGVARBINARY`.

(The same thing is true for `Types.SMALLINT`, `Types.TINYINT`, `Types.BOOLEAN`, `Types.FLOAT`, `Types.Numeric`, `Types.CHAR`, `Types.VARCHAR`, `Types.LONGNVARCHAR`, `Types.NCHAR`, `Types.NVARCHAR`, and `Types.Real`, actually).

Reply:
No, it does not matter technically until we change any mapping. But it makes sense to at least have the same behaviour for all types with either only providing a value for the SQL type we support or always try to get the value with the getter for the "group". Otherwise, I feel like it opens the door for questions about why treating String related types differently than binary related type? Even if the answer is "Why not" :P

Reply:
good enough haha, updated!

Reply:
thanks :)